yt_dlp/extractor/crunchyroll.py

```diff
@@ -97,6 +103,7 @@ def _update_auth(self):
                     'and your browser\'s User-Agent (with --user-agent)', expected=True)
             raise
 
+        CrunchyrollBaseIE._IS_PREMIUM = 'cr_premium' in jwt_decode_hs256(auth_response['access_token']).get('benefits', [])
```

Comment:
assuming `benefits` is a list/array, I would do this
```suggestion
        CrunchyrollBaseIE._IS_PREMIUM = 'cr_premium' in traverse_obj(auth_response, ('access_token', {jwt_decode_hs256}, 'benefits', ...))
        ```